docs/architecture.md

```diff
@@ -148,26 +148,45 @@ Execution will proceed similarly as in the Java case.
 
 Each option consists of a `String` unless otherwise noted.
 
-- `spark.armada.Executor.trackerPollingInterval` - Specifies
-    the interval between polls to check the state of Executors.
-- `spark.armada.Executor.trackerTimeout` - Specifies the time to
-    wait for the minimum number of Executors.
+
+- ` spark.armada.container.image` - Container image to use for Spark containers.
+- ` spark.armada.driver.limit.cores` - Specify the hard cpu limit for the driver pod
+- ` spark.armada.driver.request.cores` - Specify the cpu request for the driver pod
+- ` spark.armada.driver.limit.memory` - Specify the hard memory limit for the driver pod
+- ` spark.armada.driver.request.memory` - Specify the memory request for the driver pod
+- ` spark.armada.executor.limit.cores` - Specify the hard cpu limit for each executor pod
+- ` spark.armada.executor.request.cores` - Specify the cpu request for each executor pod
+- ` spark.armada.executor.limit.memory` - Specify the hard memory limit for each executor pod
+- ` spark.armada.executor.request.memory` - Specify the memory request for each executor pod
+- ` spark.armada.executor.connectionTimeout` - Time to wait for the executor to connect to the driver. (Long seconds)
+- ` spark.armada.queue` - The name of the job queue to use for the Armada job.
+- ` spark.armada.jobSetId` - The JobSet ID for which the driver and executor pods will be part of. 
+          If not set, it will be derived from the Spark application name.
+- ` spark.armada.internalUrl` - The Kubernetes DNS or IP address of the Armada Server.
+          This URL is used by the Driver when running in Cluster mode.
+          If not specified, 'spark.master' will be used.
+- ` spark.armada.scheduling.namespace` - The namespace to use for the job. If not set, the default namespace will be used.
+- ` spark.armada.scheduling.priority` - The priority to use for the job. If not set, the default priority will be used. (Double)
+- `spark.armada.executor.trackerPollingInterval` - Specifies
+    the interval between polls to check the state of Executors. (Long milliseconds)
```

Comment:
Spark allows to define time units like `ms`, `s` via `.timeConf`. Can we reuse that feature here?

Reply:
Actually it does use a timeconf:  https://github.com/armadaproject/armada-spark/blob/master/src/main/scala/org/apache/spark/deploy/armada/Config.scala#L63

The doc is incorrect, I'll fix it.